error_test.go

```diff
@@ -4,42 +4,33 @@ import (
 	"errors"
 	"fmt"
 	"net"
-	"reflect"
 	"testing"
 )
 
 func TestIsAsError(t *testing.T) {
-	var e Error
 	tcs := []struct {
-		err            error
-		expected       string
-		expectedTarget error
+		err      error
+		expected string
 	}{
 		{
-			err:            fmt.Errorf("%w sample error: %v", ErrInvalidVersion, 123),
-			expected:       "uuid: sample error: 123",
-			expectedTarget: &e,
+			err:      fmt.Errorf("%w sample error: %v", ErrInvalidVersion, 123),
+			expected: "uuid: sample error: 123",
 		},
 		{
-			err:            fmt.Errorf("%w", ErrInvalidFormat),
-			expected:       "uuid: invalid UUID format",
-			expectedTarget: ErrInvalidFormat,
+			err:      fmt.Errorf("%w", ErrInvalidFormat),
+			expected: "uuid: invalid UUID format",
 		},
 		{
-			err:            fmt.Errorf("%w %q", ErrIncorrectFormatInString, "test"),
-			expected:       "uuid: incorrect UUID format in string \"test\"",
-			expectedTarget: ErrIncorrectFormatInString,
+			err:      fmt.Errorf("%w %q", ErrIncorrectFormatInString, "test"),
+			expected: "uuid: incorrect UUID format in string \"test\"",
 		},
 	}
 	for i, tc := range tcs {
 		t.Run(fmt.Sprintf("Test case %d", i), func(t *testing.T) {
 			var e2 Error
-			if !errors.Is(tc.err, &e2) {
+			if !errors.As(tc.err, &e2) {
```

Comment:
This should still be testing that `errors.Is()` works correctly, it just needs to use the specific sentinel errors to do so.

I would add another `expectedErr Error` field to your test case struct, set to the wrapped sentinel error, and test it using something like this:
```
if !errors.Is(err, tc.expectedErr) {
    t.Errorf("expected error to be, or wrap, the %v sentinel error", tc.expectedErr)
}
```
As it's written right now we're testing `errors.As()` twice.